packages/client/hmi-client/src/components/workflow/ops/simulate-ciemss/tera-simulate-node-ciemss.vue

```diff
@@ -105,20 +108,46 @@ const chartProxy = chartActionsProxy(props.node, (state: SimulateCiemssOperation
 	emit('update-state', state);
 });
 
-const processResult = (runId: string) => {
+const processResult = async (runId: string) => {
 	const state = _.cloneDeep(props.node.state);
 	if (state.chartConfigs.length === 0) {
 		chartProxy.addChart();
 	}
 
+	// FIXME: Test summarization
+	const summaryStr = await getRunResult(runId, 'result_summary.csv');
+	const summaryData = csvParse(summaryStr);
+	const start = _.first(summaryData);
+	const end = _.last(summaryData);
+
+	const prompt = `
+The following are the key attributes of a simulation/forecasting process for a ODE epidemilogy model.
+
+The input parameters are as follows:
+- samples: ${state.numSamples}
+- method: ${state.method}
+- timespan: ${JSON.stringify(state.currentTimespan)}
+
+The output has these metrics at the start:
+- ${JSON.stringify(start)}
+
+The output has these metrics at the end:
+- ${JSON.stringify(end)}
+
+Provide a summary in 100 words or less.
+	`;
```

Comment:
Why this is done on the front-end?

Reply:
Because the API-side does not know about operators, and that was sort of intentional by design.

The original conception of the workflow operators was that you can extend it, eg adding a new operator, without needing to modify the core Terarium's Java code, and that the new operator is self-sufficient in someway. 

So you can do something like:
```
import { Workflow } from '@terarium/workflow`
```
and you can start building your own thing, the API provides some structure for nodes/edges but does not dictate the logic.

TLDR: the hope was to make this reusable and domain agnostic, to that end we have succeeded in some areas and kind of murky in others. But we have been following this pattern so far and haven't had any major problems yet.

Reply:
That makes complete sense!